LDEventSource/LDDataAccumulator.m

```diff
@@ -0,0 +1,31 @@
+//
+//  LDDataAccumulator.m
+//  DarklyEventSource
+//
+//  Created by Mark Pokorny on 5/30/18. +JMJ
+//  Copyright © 2018 Catamorphic Co. All rights reserved.
+//
+
+#import "LDDataAccumulator.h"
+
+@implementation LDDataAccumulator
+-(void)accumulateDataWithString:(NSString*)dataString {
+    if (dataString.length == 0) { return; }
+    if (self.dataString == nil) {
+        self.dataString = dataString;
+        return;
+    }
+    self.dataString = [self.dataString stringByAppendingString:dataString];
+}
+
+-(BOOL)isReadyToParseEvent {
+    if (self.dataString.length == 0) { return NO; }
+    NSArray<NSString*> *lines = [self.dataString componentsSeparatedByCharactersInSet:[NSCharacterSet newlineCharacterSet]];
+    if (lines.count == 0) { return NO; }
+    return [lines lastObject].length == 0;
```

Comment:
I'm a little suspect of this method. The problem we have is that the string comes in multiple chunks that are potentially incomplete. As such, it's entirely possible that we'll get a string that ends in `\n` but that isn't the end of the object, because the next character is still forthcoming, not because the object is complete.

Reply:
Using https://github.com/launchdarkly/okhttp-eventsource/blob/master/src/main/java/com/launchdarkly/eventsource/EventParser.java#L41 as a reference implementation, we need to chunk messages by empty lines (e.g. `\n\n`). Does that do this?